common/lib/common-utils/src/test/mocha/idleTaskScheduler.spec.ts

```diff
@@ -0,0 +1,66 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { strict as assert } from "assert";
+import { SinonFakeTimers, useFakeTimers } from "sinon";
+import * as idleTask from "../../idleTaskScheduler";
+
+describe("Idle task scheduler", () => {
+    let clock: SinonFakeTimers;
+
+    before(() => {
+        clock = useFakeTimers();
+    });
+
+    afterEach(() => {
+        clock.reset();
+    });
+
+    after(() => {
+        clock.restore();
+    });
+
+    function someTask(x: number): boolean {
+        return x > 3 ? true : false;
+    }
+
+    it("Should schedule and run a synchronous task during idle time", async () => {
+        let success = false;
+        (globalThis as any).requestIdleCallback = idleTask.scheduleIdleTask;
+        await new Promise((resolve, reject) => {
+            try {
+                resolve(async () => {
+                    requestIdleCallback(() => {
+                        someTask(5);
+                    });
+                });
+            } catch (e) {
+                reject(e);
+            }
+        }).then(() => {
+            success = true;
+        });
+        assert(success);
+    });
+
+    it("Should fall back to setTimeout when idle Task API is not available", async () => {
+        let success = false;
+        await new Promise((resolve, reject) => {
+            try {
+                resolve(async () => {
+                    await idleTask.scheduleIdleTask(() => {
+                        someTask(5);
+                    }, 1000);
+                    clock.tick(1100);
```

Comment:
I believe this line does not do anything. This line will move the clock forward by 1.1 seconds, but we already awaited for  scheduleIdleTask() above, so we already wasted 1 second (i.e. wait on a clock already happened, and because there was no clock.tick() call during that time, we waited for real 1.1 seconds).

You want something like that
```ts
const promise = idleTask.scheduleTask(...);
clock.tick(1100);
await promise;
```
I.e. you move time forward while there is a timer in the system, not after timer already fired